android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboard.java

```diff
@@ -443,26 +443,42 @@ public void onDestroy() {
     dismissHelpBubble();
   }
 
+  @Override
   public void onConfigurationChanged(Configuration newConfig) {
     super.onConfigurationChanged(newConfig);
 
     RelativeLayout.LayoutParams params = KMManager.getKeyboardLayoutParams();
+    // I suspect this is the part we should actually be calling directly...
     this.setLayoutParams(params);
+    this.invalidate();
+    this.requestLayout();
 
+    this.dismissHelpBubble();
+
+    if(this.getShouldShowHelpBubble()) {
+      this.showHelpBubbleAfterDelay(2000);
+    }
+  }
+
+  @Override
+  public void onSizeChanged(int width, int height, int oldWidth, int oldHeight) {
+    super.onSizeChanged(width, height, oldWidth, oldHeight);
     int bannerHeight = KMManager.getBannerHeight(context);
     int oskHeight = KMManager.getKeyboardHeight(context);
+
+    if(bannerHeight + oskHeight != height) {
+      // We'll proceed, but cautiously and with logging.
+      KMLog.LogInfo(TAG, "Height mismatch: onSizeChanged = " + height + ", our version = " + (bannerHeight + oskHeight));
+    }
+
     if (this.htmlBannerString != null && !this.htmlBannerString.isEmpty()) {
       setHTMLBanner(this.htmlBannerString);
     }
+
     loadJavascript(KMString.format("setBannerHeight(%d)", bannerHeight));
-    loadJavascript(KMString.format("setOskWidth(%d)", newConfig.screenWidthDp));
+    loadJavascript(KMString.format("setOskWidth(%d)", width));
+    // Must be last - it's the one that triggers a Web-engine layout refresh.
     loadJavascript(KMString.format("setOskHeight(%d)", oskHeight));
```

Comment:
This smells like it should be a single function call, `setDimensions({bannerHeight:%d,oskWidth:%d,oskHeight:%d})`.

This may also be relevant:
https://github.com/keymanapp/keyman/blob/91e91bf91b56efb6cc20ebabd4a6f0a257801af0/android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboard.java#L426-L436 

Calls to these functions (on master branch):
* Only 1 call to `setBannerHeight`, right here.
* 4 calls to `setOskHeight`
* 2 calls to `setOskWidth`